ahcore/transforms/pre_transforms.py

```diff
@@ -117,6 +124,38 @@ def __repr__(self) -> str:
         return f"PreTransformTaskFactory(transforms={self._transforms})"
 
 
+class SampleNFeatures:
+    def __init__(self, n: int = 1000) -> None:
+        self.n = n
+        logger.warning(
+            f"Sampling {n} features from the image. Sampling WITH replacement is done if there are not enough tiles."
+        )
+
+    def __call__(self, sample: DlupDatasetSample) -> DlupDatasetSample:
+        features = sample["image"]
+
+        # Get the dimensions of the image
+        h = features.height  # Height
+        w = features.width  # Width
+
+        if h != 1:
+            raise ValueError(f"Expected features to have a width dimension of 1, got {h}.")
+
+        n_random_indices = (
+            np.random.choice(w, self.n, replace=False) if w > self.n else np.random.choice(w, self.n, replace=True)
+        )
+
+        # Extract the selected columns (indices) from the image
+        # Create a new image from the selected indices
+        # todo: this can probably be done without a for-loop quicker
+        selected_columns = [features.crop(idx, 0, 1, h) for idx in n_random_indices]
```

Comment:
you need to seed on something to make sure that each run gives reproducible values. Maybe the path is a good seed?

Reply:
I think the seed_everything function in the entrypoints should handle this right?